core/src/main/resources/jenkins/model/Jenkins/login.jelly

```diff
@@ -23,58 +23,120 @@ THE SOFTWARE.
+    <body>
+      <j:choose>
+        <j:when test="${it.setupWizard.isUsingSecurityToken()}">
+          <st:include it="${it.setupWizard}" page="authenticate-security-token"/>
+        </j:when>
+        <j:otherwise>
+          <div class="login-main">
+            <div class="modal">
+              <st:include it="${loginDecorator}" page="login-header.jelly" optional="true"/>
+              <!-- login form -->
+              <form name="login" action="${it.securityRealm.authenticationGatewayUrl}" method="post">
+                <j:if test="${it.securityRealm.allowsSignup()}">
+                  <div style="margin-top:2em">
+                    ${%signUp}
+                  </div>
+                </j:if>
+                <j:if test="${error}">
+                  <div class="alert alert-danger">
+                    ${%Invalid username or password}
+                  </div>
+                </j:if>
+                <div>
+                  <input
+                    class="${inputClass}"
+                    type="text"
+                    name="j_username"
+                    id="j_username"
+                    autocorrect="off"
+                    autocapitalize="off"
+                    placeholder="${%User}"
+                  />
+                </div>
+                <div>
+                  <input
+                    class="${inputClass}"
+                    type="password"
+                    name="j_password"
+                    placeholder="${%Password}"
+                  />
+                </div>
+                <input type="hidden" name="from" value="${from}" />
+                <div class="submit"><f:submit value="${%signIn}" /></div>
+                <!-- focus the username field -->
+                <script type="text/javascript">
+                  document.getElementById('j_username').focus();
+                  var checkBoxClick = function(event) {
+                    document.getElementById('remember_me').click();
+                  }
+                </script>
 
-        <j:forEach var="fls" items="${it.getFederatedLoginServices()}">
-          <div>
-            <st:include page="loginFragment.jelly" it="${fls}"/>
-          </div>
-        </j:forEach>
+                <j:if test="${app.security.name()=='SECURED' and !app.disableRememberMe}">
+                  <!-- remember me service only available for Acegi -->
+                  <div class="checkboxWrapper">
+                    <div class="checkbox">
+                      <input type="checkbox" id="remember_me" name="remember_me" />
+                      <label for="remember_me"> </label>
+                    </div>
+                    <span onClick="checkBoxClick()">${%Keep me signed in}</span>
+                  </div>
+                </j:if>
+
+              </form>
+
+              <j:forEach var="fls" items="${it.getFederatedLoginServices()}">
+                <div>
+                  <st:include page="loginFragment.jelly" it="${fls}"/>
```

Comment:
Will this actually continue to work as expected with the redesign?

Reply:
Only existing implementations in https://plugins.jenkins.io/openid and https://plugins.jenkins.io/wwpass-plugin (this one can probably be ignored)

Reply:
they should work as before

Reply:
Right, but possibly need followup design changes to adapt presentation?

Reply:
They may want to adopt their css yeah but https://github.com/jenkinsci/openid-plugin/blob/master/src/main/resources/hudson/plugins/openid/OpenIdLoginService/loginFragment.jelly is providing inline styles, which would be applied.

Reply:
So what you're saying is that merging your change will result in broken presentation of the login page downstream in OpenID plugin?

Reply:
I just tried but that plugin is broken, anyway all should work as before.